x-pack/plugins/canvas/public/state/actions/elements.js

```diff
@@ -263,7 +263,17 @@ function setExpressionFn({ dispatch, getState }, expression, elementId, pageId,
 
   // read updated element from state and fetch renderable
   const updatedElement = getNodeById(getState(), elementId, pageId);
-  if (doRender === true) {
+
+  // reset element.filter if element is no longer a filter
+  if (
+    updatedElement.filter &&
+    !['dropdownControl', 'timefilterControl', 'exactly'].some(filter =>
```

Comment:
Is there a way we can get the list of filter elements without hard-coding them here? Maybe we can pull the functions from the registry somehow?

Reply:
I'd need to somehow check for render functions that specifically render filters. I don't see anything in the renderer def that I could use to single out filter renderers.

I could introduce a new property in the renderer def like `subtype: filter`, but I think this should be addressed in a separate PR.

Reply:
Yeah, I gave this some thought too, and couldn't come up with anything simple here. I'm cool derfering on this as well, just please open an issue for it.

Reply:
I added a TODO to that line, and I'll open an issue.